src/Commands/Migrate.php

```diff
@@ -42,17 +46,24 @@ public function handle(): int
             return 1;
         }
 
-        tenancy()->runForMultiple($this->getTenants(), function ($tenant) {
-            $this->components->info("Tenant: {$tenant->getTenantKey()}");
+        try {
+            tenancy()->runForMultiple($this->option('tenants'), function ($tenant) {
+                $this->line("Tenant: {$tenant->getTenantKey()}");
 
-            event(new MigratingDatabase($tenant));
+                event(new MigratingDatabase($tenant));
+                // Migrate
+                parent::handle();
 
-            // Migrate
-            parent::handle();
+                event(new DatabaseMigrated($tenant));
+            });
 
-            event(new DatabaseMigrated($tenant));
-        });
+            return 0;
+        } catch (TenantDatabaseDoesNotExistException|QueryException $th) {
+            if (! $this->option('skip-failing')) {
+                throw $th;
+            }
 
-        return 0;
+            return 1;
```

Comment:
Wouldn't this basically terminate at the first failing tenant and then not continue? Even if the exception is suppressed.

Reply:
I see, yeah, makes sense that would happen. I'll see if I can add a test for this and make it work correctly

Reply:
The test looks good. I think the `try {}` should be inside the loop basically, not outside.

Reply:
I tried that, but since `runForMultiple` throws the exception (not the closure passed to it), `try ()` inside the closure doesn't catch the error

Reply:
I think the simplest way to make this work would be calling `$tenant->run()` one by one (the test passes with that apporach). Would that hinder the performance a lot?

```php
foreach ($tenants as $tenant) {
    try {
        $tenant->run(function()) {
            // Code...
        }};
    } catch {
        if (! $this->option('skip-failing')) {
            throw $th;
        }
    }
}
```

Reply:
`runForMultiple()` is a tiny bit more efficient than `foreach()` & `run()`, but the difference is negligible so this doesn't matter in a console command.